src/validate.rs

```diff
@@ -263,9 +263,20 @@ fn check_link(
         }
     }
 
+    let mut request = client.get(&url);
+
+    for (pattern, headers) in cfg.http_headers.iter() {
```

Comment:
Instead of using `Client::get()`, is it worth using `RequestBuilder` directly then pulling request creation out into its own function?

That way we can add a `additional_headers()` method to the `Config` which will match each pattern against the URL and return `impl Iterator<Item=(http::HeaderName, http::HeaderValue)>` which can be `collect()`-ed into a `http::HeaderMap` for the [`RequestBuilder::headers()`](https://docs.rs/reqwest/0.10.0/reqwest/struct.RequestBuilder.html#method.headers) method... Or is that maybe overcomplicating things?

Reply:
I'll see into it tomorrow. My first thought is - "good idea!".

Reply:
You know, the API you described is quite sensible yet over complicated. For now I decided to leave things as is, we can add additional API on demand anyway.